x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -8,49 +8,41 @@ import _ from 'lodash';
 import React, { useState, useEffect, useCallback, useRef } from 'react';
 import { I18nProvider } from '@kbn/i18n/react';
 import { i18n } from '@kbn/i18n';
-import { EuiLink, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 import { Storage } from 'ui/storage';
 import { CoreStart, SavedObjectsClientContract } from 'src/core/public';
+import {
+  DataSetup,
+  IndexPattern as IndexPatternInstance,
+  SavedQuery,
+} from 'src/legacy/core_plugins/data/public';
+import { Filter } from '@kbn/es-query';
+import { TopNavMenu } from '../../../../../../src/legacy/core_plugins/kibana_react/public';
 import { Query } from '../../../../../../src/legacy/core_plugins/data/public/query';
-import { QueryBarTopRow } from '../../../../../../src/legacy/core_plugins/data/public/query/query_bar';
 import { Document, SavedObjectStore } from '../persistence';
 import { EditorFrameInstance } from '../types';
 import { NativeRenderer } from '../native_renderer';
 
 interface State {
   isLoading: boolean;
   isDirty: boolean;
+  // indexPatternTitles: string[];
+  indexPatterns: IndexPatternInstance[];
+  persistedDoc?: Document;
+
+  // Properties needed to interface with TopNav
   dateRange: {
     fromDate: string;
     toDate: string;
   };
   query: Query;
-  indexPatternTitles: string[];
-  persistedDoc?: Document;
-  localQueryBarState: {
-    query?: Query;
-    dateRange?: {
-      from: string;
-      to: string;
-    };
-  };
-}
-
-function isLocalStateDirty(
-  localState: State['localQueryBarState'],
-  query: Query,
-  dateRange: State['dateRange']
-) {
-  return Boolean(
-    (localState.query && query && localState.query.query !== query.query) ||
-      (localState.dateRange && dateRange.fromDate !== localState.dateRange.from) ||
-      (localState.dateRange && dateRange.toDate !== localState.dateRange.to)
-  );
+  filters: Filter[];
+  savedQuery?: SavedQuery;
 }
 
 export function App({
   editorFrame,
   core,
+  data,
```

Comment:
I would recommend using DataStart here. At the moment there's no big difference, but there will be in the near future.

Reply:
As discussed we don't have access to dataStart yet because it's not merged